src/Writer/XLSX/Options/WorkbookProtection.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenSpout\Writer\XLSX\Options;
+
+final class WorkbookProtection
+{
+    private ?string $passwordHash = null;
+    private bool $lockStructure = false;
+    private bool $lockWindows = false;
+    private bool $lockRevisions = false;
+
+    /**
+     * @return $this
+     */
+    public function setPassword(string $password): self
+    {
+        $this->passwordHash = $this->createPasswordHash($password);
+
+        return $this;
+    }
+
+    /**
+     * @return $this
+     */
+    public function setLockStructure(bool $lockStructure): self
+    {
+        $this->lockStructure = $lockStructure;
+
+        return $this;
+    }
+
+    /**
+     * @return $this
+     */
+    public function setLockWindows(bool $lockWindows): self
+    {
+        $this->lockWindows = $lockWindows;
+
+        return $this;
+    }
+
+    /**
+     * @return $this
+     */
+    public function setLockRevisions(bool $lockRevisions): self
+    {
+        $this->lockRevisions = $lockRevisions;
+
+        return $this;
+    }
+
+    public function getXml(): string
+    {
+        return '<workbookProtection'.$this->getSheetViewAttributes().'/>';
+    }
+
+    private function createPasswordHash(string $password): string
```

Comment:
This seems copy-pasted from `SheetProtection`.
Please:

1. Move the hashing function into a separate class
2. Add a specific test with hardcoded password and hash result